pkg/apis/core/v1beta1/defaults.go

```diff
@@ -278,6 +278,19 @@ func SetDefaults_Shoot(obj *Shoot) {
 
 		obj.Spec.Provider.Workers[i].CRI = &CRI{Name: CRINameContainerD}
 	}
+
+	if obj.Spec.SystemComponents == nil {
+		obj.Spec.SystemComponents = &SystemComponents{}
+	}
+	if obj.Spec.SystemComponents.CoreDNS == nil {
+		obj.Spec.SystemComponents.CoreDNS = &CoreDNS{}
+	}
+	if obj.Spec.SystemComponents.CoreDNS.Autoscaling == nil {
+		obj.Spec.SystemComponents.CoreDNS.Autoscaling = &CoreDNSAutoscaling{}
+	}
+	if obj.Spec.SystemComponents.CoreDNS.Autoscaling.Mode != CoreDNSAutoscalingModeHorizontal && obj.Spec.SystemComponents.CoreDNS.Autoscaling.Mode != CoreDNSAutoscalingModeClusterProportional {
+		obj.Spec.SystemComponents.CoreDNS.Autoscaling.Mode = CoreDNSAutoscalingModeHorizontal
+	}
```

Comment:
Can you please add a unit test for this defaulting (same in `v1alpha1`)?

Reply:
Added test with newest commit for v1beta1 as well as for v1alpha1.